docs/design.md

```diff
@@ -98,30 +108,36 @@ By returning `EntityCommands`, you can easily chain multiple widgets together an
 Define your assets in an enum so each variant maps to a `Handle`:
 
 ```rust
-#[derive(Copy, Clone, Eq, PartialEq, Hash, Reflect)]
-pub enum SpriteKey {
-    Player,
-    Enemy,
-    Powerup,
-}
-
-impl AssetKey for SpriteKey {
-    type Asset = Image;
+#[derive(Resource, Debug, Deref, DerefMut, Reflect)]
+#[reflect(Resource)]
+pub struct ImageHandles(HashMap<String, Handle<Image>>);
+
+impl ImageHandles {
+    pub const KEY_PLAYER: &'static str = "Player";
+    pub const KEY_ENEMY: &'static str = "Enemy";
+    pub const KEY_POWERUP: &'static str = "Powerup";
 }
 
-impl FromWorld for HandleMap<SpriteKey> {
+impl FromWorld for ImageHandles {
     fn from_world(world: &mut World) -> Self {
         let asset_server = world.resource::<AssetServer>();
-        [
-            (SpriteKey::Player, asset_server.load("player.png")),
-            (SpriteKey::Enemy, asset_server.load("enemy.png")),
-            (SpriteKey::Powerup, asset_server.load("powerup.png")),
-        ]
-        .into()
+        let map = [(
+            ImageHandles::KEY_PLAYER,
+            asset_server.load("player.png"),
+        ), (
+            ImageHandles::KEY_ENEMY,
+            asset_server.load("enemy.png"),
+        ), (
+            ImageHandles::KEY_POWERUP,
+            asset_server.load("powerup.png"),
+        )]
+        .into();
+        Self(map)
     }
 }
 ```
 
+TODO
```

Comment:
Intentional?

Reply:
Yes, as this is still a draft and I wanted to know where I left off :D